third_party/terraform/utils/common_operation.go

```diff
@@ -96,6 +97,11 @@ func CommonRefreshFunc(w Waiter) resource.StateRefreshFunc {
 		if err != nil {
 			// Importantly, this error is in the GET to the operation, and isn't an error
 			// with the resource CRUD request itself.
+			if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
```

Comment:
if you want you can use isGoogleApiErrorWithCode(err, 404)

Reply:
I think I prefer this because it's closer to https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/utils/utils.go#L353